codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -669,12 +671,21 @@ public void operationComplete(ChannelFuture future) throws Exception {
     public ChannelFuture goAway(final ChannelHandlerContext ctx, final int lastStreamId, final long errorCode,
                                 final ByteBuf debugData, ChannelPromise promise) {
         try {
+            promise = promise.unvoid();
             final Http2Connection connection = connection();
-            if (connection.goAwaySent() && lastStreamId > connection.remote().lastStreamKnownByPeer()) {
-                throw connectionError(PROTOCOL_ERROR, "Last stream identifier must not increase between " +
-                                                      "sending multiple GOAWAY frames (was '%d', is '%d').",
-                                      connection.remote().lastStreamKnownByPeer(), lastStreamId);
+            if (connection().goAwaySent()) {
```

Comment:
I added a check to protect against re-entrancy, which could happen if sending a GO_AWAY fails, as this would now trigger error handling, which would attempt to send a GO_AWAY ...

I am not sure if it could have happend before, but defenitely now that we listen for errors on GO_AWAY. Should I open an issue and close it immediately?